src/main/java/org/dcsa/skernel/repositority/PartyContactDetailsRepository.java

```diff
@@ -4,10 +4,26 @@
 import org.springframework.data.repository.reactive.ReactiveCrudRepository;
 import org.springframework.stereotype.Repository;
 import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
 
 import java.util.UUID;
 
 @Repository
-public interface PartyContactDetailsRepository extends ReactiveCrudRepository<PartyContactDetails, UUID> {
+public interface PartyContactDetailsRepository
+    extends ReactiveCrudRepository<PartyContactDetails, UUID> {
   Flux<PartyContactDetails> findByPartyID(String partyID);
+
+  Mono<PartyContactDetails> findByNameAndEmailAndAndPhoneAndUrl(
+      String name, String email, String phone, String url);
+
+  default Mono<PartyContactDetails> findByContent(PartyContactDetails partyContactDetails) {
+    if (partyContactDetails.getId() != null) {
+      return findById(partyContactDetails.getId());
+    }
+    return findByNameAndEmailAndAndPhoneAndUrl(
+        partyContactDetails.getName(),
+        partyContactDetails.getEmail(),
+        partyContactDetails.getPhone(),
+        partyContactDetails.getUrl());
+  }
```

Comment:
Is this safe given that the party contact details in the IM has a `party_id` linking to the concrete party?

What if two different party had the same contact details (because the party had to be different for another reason)?